### PR TITLE
Added The return of Post Value when an Integration Exception is thrown while Save Action

### LINKED ISSUE #28143: [Issue] Added The return of Post Value when an Integration Exception is thrown while Save Action
This issue is automatically created based on existing pull request: magento/magento2#26660: Added The return of Post Value when an Integration Exception is thrown while Save Action

---------
This keeps the form data persistent when the an Integration Exception is throw as a case when name of Integration being added is same.

### Preconditions (*)
Magento 2.4-develop

### Steps to reproduce (*)

- Go to System-> Integrations 
- Add a new Integration , Choose a name that already exist and fill all the form and then hit Save.

### Expected result (*)
The Save Actions Fails with a message The integration with name "%1" exists. but the form entries are **intact**.

### Actual result (*)
The Save Actions Fails with a message The integration with name "%1" exists. but the form entries are **reset**.

### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by realadityayadav**
This keeps the form data persistent when the an Integration Exception is throw as a case when name of Integration being added is same.

### Description (*)

Magento 2.x - This issue reproduce in all version.

Go to System-> Integrations 
Add a new Integration , Choose a name that already exist and fill all the form and then hit Save.

Expected - The Save Actions Fails with a message The integration with name "%1" exists. but the form entries are intact.
Actual - The Save Actions Fails with a message The integration with name "%1" exists. but the form entries are reset.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#28143: Added The return of Post Value when an Integration Exception is thrown while Save Action

### CODE REVIEW

**Comment by realadityayadav** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @realadityayadav. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @realadityayadav, here is your [new Magento instance](https://pr-26660.instances.magento-community.engineering).
Admin access: [https://pr-26660.instances.magento-community.engineering/admin_60db](https://pr-26660.instances.magento-community.engineering/admin_60db)
Login: `bd4fb776` Password: `7d8c102c5bff`
Instance will be terminated in up to 3 hours.

**Comment by rogyar** [comment]
Hi @realadityayadav. Thank you for your collaboration. 
According to the new [Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all changes should be covered by automated tests. Could I kindly ask you to cover your changes with an MFTF test which will attempt to create another integration with the same name and assert that the form data persists? 

Thank you!

**Comment by realadityayadav** [comment]
Ya why not. Could you guide me to a reference of adding an MFTF please?

**Comment by sivaschenko** [comment]
Probably the last step verifying the form field should be added to this test: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Integration/Test/Mftf/Test/AdminCreateIntegrationEntityWithDuplicatedNameTest.xml

**Comment by realadityayadav** [comment]
@rogyar The Test are failing for a different module in the EE & B2B files which are not yet available in this release?
Can you take a look once?

File: app/code/Magento/Integration/Test/Mftf/Test/AdminCreateIntegrationEntityWithDuplicatedNameTest.xml [hunk 1]
```diff
@@ -46,5 +46,6 @@
             <argument name="message" value="The integration with name &quot;Integration1&quot; exists."/>
             <argument value="error" name="messageType"/>
         </actionGroup>
+        <seeInField stepKey="checkEnteredValueIsPreserved" selector="{{AdminNewIntegrationSection.name}}" userInput="Integration1"/>
```
Comment (rogyar) [inline app/code/Magento/Integration/Test/Mftf/Test/AdminCreateIntegrationEntityWithDuplicatedNameTest.xml]: Hi @realadityayadav. According to the [best practices](https://github.com/magento/magento-functional-tests-migration/wiki/Best-practices) it's highly recommended to build a test using action groups without direct operations in the test body. As you can see from the test you have edited, it consists of the action groups.

Please, move this check into a new action group and name the action group as `AssertAdminIntegrationNameInFormActionGroup`.  Then you can use the action group in your test.

Thank you!

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
Hi @realadityayadav. Looks like you forgot to create the mentioned action group. According to my previous comment you need to perform the following steps. 

1. Create a new action group (`AssertAdminIntegrationNameInFormActionGroup`)
2. Move `seeInField` check into the newly created action group
3. Use the newly created action group in the test body.

As far as I see, steps 1-2 are missing so far.

File: app/code/Magento/Integration/Test/Mftf/Test/AdminCreateIntegrationEntityWithDuplicatedNameTest.xml (hunk 1 above)
Comment (realadityayadav) [reply app/code/Magento/Integration/Test/Mftf/Test/AdminCreateIntegrationEntityWithDuplicatedNameTest.xml]: https://github.com/magento/magento2/pull/26660/files

Please see the recent change now @rogyar

**Comment by realadityayadav** [comment]
I think it is already there. Please check the latest version.

**Comment by rogyar** [comment]
Hi @realadityayadav. You have 2 files modified in your PR: the actual fix and the test. No action group so far.

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7523](https://jira.corp.magento.com/browse/ENGCOM-7523) has been created to process this Pull Request

**Comment by realadityayadav** [comment]
Thanks @engcom-Charlie for covering the test cases. I was not very familiar with this & @sivaschenko @rogyar for the help on the pull request.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**
1. Go to Admin->System->Integrations;
2. Add a new Integration;
3. Choose a name that already exists and fill all the form ;
4. Save;

**Before:** :heavy_multiplication_x: The form entries are reset.

Peek 2020-05-07 09-43

**After:** :heavy_check_mark: The form entries are intact.

Peek 2020-05-07 09-46

**Comment by slavvka** [comment]
@magento create issue

**Comment by slavvka** [comment]
@magento run WebAPI tests

**Comment by slavvka** [comment]
@magento run WebAPI Tests

**Comment by slavvka** [comment]
@magento run Integration Tests

**Pull request merged**